library/pk_wrap.c

```diff
@@ -522,7 +522,7 @@ static size_t eckey_get_bitlen(mbedtls_pk_context *pk)
 }
 
 #if defined(MBEDTLS_PK_CAN_ECDSA_VERIFY)
-#if defined(MBEDTLS_USE_PSA_CRYPTO)
+#if defined(MBEDTLS_PSA_CRYPTO_CLIENT)
```

Comment:
Functions that allow PK to cooperate with PSA should be guarded by `MBEDTLS_PSA_CRYPTO_CLIENT` (e.g. `mbedtls_pk_setup_opaque`, `mbedtls_pk_copy_from_psa`, `mbedtls_pk_import_into_psa`, …). But functions that cause PK to dispatch to PSA instead of built-in code should remain guarded by `MBEDTLS_USE_PSA_CRYPTO`: that's what it's for.

So the PSA version of `ecdsa_verify_wrap` needs to remain gated by `MBEDTLS_USE_PSA_CRYPTO`. But `ecdsa_opaque_verify_wrap` needs to be available whenever `MBEDTLS_PSA_CRYPTO_CLIENT` is defined.

Reply:
Thanks for the explanation! This was exactly the doubt I had while working on this PR because `USE_PSA` and `CRYPTO_CLIENT` looked very very similar to me.

Reply:
Last question. It seems to me that `mbedtls_pk_wrap_as_opaque()` might be guarded with `CRYPTO_CLIENT` as well for the same reason as `setup_opaque()`, but I guess that we're ignoring this because this is going to be deprecated, right?